Patch for building OpenVPN with libressl, taken from Void Linux.

diff -upr openvpn-2.4.5.orig/src/openvpn/openssl_compat.h openvpn-2.4.5/src/openvpn/openssl_compat.h
--- openvpn-2.4.5.orig/src/openvpn/openssl_compat.h	2018-03-20 11:29:16.324926371 +0100
+++ openvpn-2.4.5/src/openvpn/openssl_compat.h	2018-03-20 11:29:22.118289384 +0100
@@ -711,7 +711,7 @@ SSL_CTX_get_max_proto_version(SSL_CTX *c
 }
 #endif /* SSL_CTX_get_max_proto_version */
 
-#ifndef SSL_CTX_set_min_proto_version
+#if !defined(SSL_CTX_set_min_proto_version) && !defined(LIBRESSL_VERSION_NUMBER)
 /** Mimics SSL_CTX_set_min_proto_version for OpenSSL < 1.1 */
 static inline int
 SSL_CTX_set_min_proto_version(SSL_CTX *ctx, long tls_ver_min)
@@ -740,7 +740,7 @@ SSL_CTX_set_min_proto_version(SSL_CTX *c
 }
 #endif /* SSL_CTX_set_min_proto_version */
 
-#ifndef SSL_CTX_set_max_proto_version
+#if !defined(SSL_CTX_set_max_proto_version) && !defined(LIBRESSL_VERSION_NUMBER)
 /** Mimics SSL_CTX_set_max_proto_version for OpenSSL < 1.1 */
 static inline int
 SSL_CTX_set_max_proto_version(SSL_CTX *ctx, long tls_ver_max)
